packages/truffle-core/lib/mnemonics/mnemonic.js

```diff
@@ -0,0 +1,74 @@
+/**
+* @module mnemonic;
+* @requires module:Configstore
+* @requires module:random
+* @requires module:bip39
+* @requires module:ethereumjs-wallet/hdkey
+*/
+
+const Configstore = require('configstore');
+const seedrandom = require('seedrandom');
+const bip39 = require('bip39'); 
+const hdkey = require('ethereumjs-wallet/hdkey');
+const crypto = require('crypto');
+
+const getOrGenerateMnemonic = {
+
+  /**
+  * gets user-level mnemonic from user config, and if missing generates a new mnemonic
+  * @returns {String} mnemonic
+  */
+  mnemonic: function () {
+    let mnemonic;
+    
+    const defaultUserConfig = new Configstore('truffle', {}, { globalConfigPath: true });
```

Comment:
Can we perhaps move `Configstore` so that it is an internal concern of the truffle-config package?

I am thinking it might be nicer if it were something like:

```javascript
const Config = require("truffle-config");
const defaultUserConfig = Config.getUserConfig();
```

This way, if we want to switch to another package besides configstore later, we won't have to change it in more than one place.